src/libcore/slice/mod.rs

```diff
@@ -2228,6 +2236,16 @@ impl<'a, T> ExactSizeIterator for Chunks<'a, T> {}
 #[unstable(feature = "fused", issue = "35602")]
 impl<'a, T> FusedIterator for Chunks<'a, T> {}
 
+#[doc(hidden)]
+unsafe impl<'a, T> TrustedRandomAccess for Chunks<'a, T> {
+    unsafe fn get_unchecked(&mut self, i: usize) -> &'a [T] {
+        let start = i * self.chunk_size;
+        let end = cmp::min(start + self.chunk_size, self.v.len());
```

Comment:
There is a potential wraparound here isn't there? i using bounds but start + size can still wrap around. In particular for `[(); !0].chunks(!0/2 +1)` I think.

Reply:
I mean that "i is in bounds"

Reply:
It turns out Rust ICES on that array. For those that don't like my mobile-friendly syntax, `!0` is the same as `usize::max_value()`.

However Rust accepts a vec of this length, and we can demo [(playground link)](https://play.rust-lang.org/?gist=122589ec765d923f36890c23dab0f0d7&version=nightly)

```rust
fn main() {
    let v = vec![(); !0];
    let iter = v.chunks((!0)/2 +1);
    for elt in iter {
        println!("Found chunk of len {}", elt.len());
    }
}
```

Which has output:

```
Found chunk of len 9223372036854775808
Found chunk of len 9223372036854775807
```